crates/core/src/rpc/protocol/v1.rs

```diff
@@ -717,6 +777,10 @@ pub trait RpcProtocolV1: RpcContext {
 	// ------------------------------
 
 	async fn query(&self, params: Array) -> Result<Data, RpcError> {
+		// Check if the user is allowed to query
+		if !self.kvs().allows_query_by_subject(self.session().au.as_ref()) {
+			return Err(RpcError::MethodNotAllowed);
+		}
```

Comment:
This seems to be a bit heavy, especially if the feature is not used

Reply:
@emmanuel-keller the capability which allows or disallows arbitrary querying for guest, record and system user-levels is not experimental

Reply:
What do you think the cost will be here, out of interest?